subsys/bluetooth/mesh/Kconfig

```diff
@@ -33,20 +33,35 @@ config BT_MESH_PROVISIONER
 	bool "Provisioner support"
 	select BT_ECC
 	select BT_MESH_PROV
+	select BT_MESH_NETWORK_MGMT
```

Comment:
You mean having a `depends on BT_MESH_PROVISIONER` under `config BT_MESH_NETWORK_MGMT`?

Currently the provisioner does actually need the database as it allocates a node and also checks if the address is available. This could be made optional but then I think there would have to be some other mechanism to either provide the device key and provision start or retrieving it once it is finished. Otherwise it wouldn't really be possible to store the device keys in another place either since it would be lost as soon as provisioning is finished. Does that make sense? :p